src/core/radio/radio.cpp

```diff
@@ -33,6 +33,58 @@
 
 namespace ot {
 
+namespace {
+#if OPENTHREAD_CONFIG_RADIO_LINK_IEEE_802_15_4_ENABLE
+constexpr Mac::ShortAddress kDefaultShortAddress = 0xfffe;
+#endif
+} // namespace
+
+void Radio::Init(void)
+{
+#if OPENTHREAD_CONFIG_RADIO_LINK_IEEE_802_15_4_ENABLE
+    Error error = OT_ERROR_NONE;
+
+    Mac::ExtAddress extAddr{};
+    Mac::Key        prevKey{};
+    Mac::Key        currKey{};
+    Mac::Key        nextKey{};
+
+    OT_UNUSED_VARIABLE(error);
+
+#if OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
+    error = EnableCsl(0, kDefaultShortAddress, nullptr);
+    OT_ASSERT(error == OT_ERROR_NONE);
+#endif
+
+    EnableSrcMatch(false);
+    ClearSrcMatchShortEntries();
+    ClearSrcMatchExtEntries();
+
+    if (IsEnabled())
+    {
+        error = Sleep();
+        OT_ASSERT(error == OT_ERROR_NONE);
+
+        error = Disable();
+        OT_ASSERT(error == OT_ERROR_NONE);
+    }
+
+    SetPanId(Mac::kPanIdBroadcast);
+    SetExtendedAddress(extAddr);
+    SetShortAddress(kDefaultShortAddress);
+    SetMacKey(0, 0, prevKey, currKey, nextKey);
+    SetMacFrameCounter(0);
+
+    error = SetTransmitPower(0);
+    OT_ASSERT(error == OT_ERROR_NONE);
+
+    error = SetCcaEnergyDetectThreshold(0);
```

Comment:
Generally, the platform layer defines a default `CcaEnergyDetectThreshold` value. The OpenThread host may not set the  `CcaEnergyDetectThreshold`, then the platform can directly use the platform defined default value. If we set the `CcaEnergyDetectThreshold` value here,  the value CcaEnergyDetectThreshold `0` will be used by the platform.  This may cause issues. The line 78 also has this problem.

Reply:
Removed.